cachalot/utils.py

```diff
@@ -178,18 +179,32 @@ def __update_annotated_subquery(_annotation: Subquery):
             else:
                 tables.update(_get_tables(db_alias, _annotation.query))
 
+        def flatten(expression: BaseExpression):
+            """
+            Recursively yield this expression and all subexpressions, in
+            depth-first order.
+
+            Taken from Django 3.2 as the previous Django versions don’t check
+            for existence of flatten.
+            """
+            yield expression
+            for expr in element.get_source_expressions():
+                if expr:
+                    if hasattr(expr, 'flatten'):
+                        yield from flatten(expr)
+                    else:
+                        yield expr
+
         # Gets tables in subquery annotations.
         for annotation in query.annotations.values():
-            if isinstance(annotation, Case):
-                for case in annotation.cases:
-                    for subquery in _find_subqueries_in_where(case.condition.children):
-                        tables.update(_get_tables(db_alias, subquery))
-                if isinstance(annotation.default, Subquery):
-                    __update_annotated_subquery(annotation.default)
-            elif isinstance(annotation, Subquery):
-                __update_annotated_subquery(annotation)
-            elif type(annotation) in UNCACHABLE_FUNCS:
+            if type(annotation) in UNCACHABLE_FUNCS:
                 raise UncachableQuery
+            for element in flatten(annotation):
+                if isinstance(element, Subquery):
+                    __update_annotated_subquery(element)
+                elif isinstance(element, RawSQL):
+                    sql = element.as_sql(None, None)[0].lower()
```

Comment:
I'd assume the compiler with the db engine is already attached, but just in case is there a method to check for db alias like in L233?

```sql = query.get_compiler(db_alias).as_sql()[0].lower()```

Reply:
Nvm I see the params are actually the compiler...